coreMacros/src/main/scala/Chisel/internal/sourceinfo/SourceInfoTransform.scala

```diff
@@ -6,7 +6,7 @@
 package Chisel.internal.sourceinfo
 
 import scala.language.experimental.macros
-import scala.reflect.macros.blackbox.Context
+import scala.reflect.macros.whitebox.Context
```

Comment:
Yes, very much so. Whitebox macros play very odd games with the typer and can make things surprising (as the type of the macro function is no longer true). The scala macros team has always been vague on how long/how they plan to support whitebox macros. 

Similarly, you lose a type check on the macro application that it at least returned the expected type. At least then you have a marker that the macro application is also toying with the type. This ensures our exposure to such changes and fancy behaviors is properly contained